scripts/typescript/afrl-to-new-api/src/scripts/afrl-csv-to-json.ts

```diff
@@ -0,0 +1,631 @@
+
+        // Create a project with the collection in it        
+        this.project = {
+            name: config.project_name,
+            model_version: MODEL_VERSION,
+            node: ['Project'],
+            collection: [this.collection],
+        } as IProject;
+
+    }
+
+    get_errors(): any {
+        return [...this.errors];
+    }
+
+    private generate_uid(): number {
+        this.last_uid += 1;
+        return this.last_uid
+    }
+
+    private get_citation(row: AFRLData): ICitation | undefined {
+
+        if(row.reference === undefined) {
+            this.record_error(`Unable to get)citation for row ${row.csv_raw_index}`)
+            return undefined;
+        }
+        // Check if citation was already created
+        const existing_citation = this.citations.get(row.reference);
+        if (existing_citation) {
+            console.log(`-- Found existing reference: ${existing_citation.reference.title}`)
+            return existing_citation;
+        }
+
+        // Create citation
+        const citation: ICitation = {
+            node: ['Citation'],
+            reference: {
+                title: row.reference,
+                type: 'database', // raw string, should be ideally picked from vocab
+                node: ['Reference'],
+                model_version: MODEL_VERSION,
+            } as IReference,
+            type: 'reference', // FIXME: typings are incorrect, should be a string.
+            model_version: MODEL_VERSION,
+        } as ICitation;
+
+        // get DOI and authors
+        const DOI_DOT_ORG = "doi.org/";
+        if (row.reference.includes(DOI_DOT_ORG)) {
+            citation.reference.doi = row.reference.replace(DOI_DOT_ORG, "");
+        } else {
+            // Putting the whole text as an author
+            citation.reference.author = [row.reference];
+
+            // The following does not work, but a better version can work.
+            // I am unsure the reference is always formatted the same way.
+            // citation.reference.author = row.reference.split(',')            // Authors are usually splitted by comas.
+            //                                          .map( a => a.trim() ); // remove empty pre/post spaces
+        }
+
+        // Store in hashmap
+        this.citations.set(row.reference, citation);
+
+        return citation;
+
+    }
+
+    private get_solvent(row: AFRLData): IMaterial {
+
+        const cas = row.solvent_CAS.trim(); // will be used as key in the hashmap this.solvent
+
+        // Try to reuse an existing solvent
+        const existing_solvent = this.solvents.get(cas);
+        if (existing_solvent) {
+            console.log(`-- Found existing solvent: ${existing_solvent.name} (cas: ${cas})`)
+            return existing_solvent
+        }
+
+        // Pull solvent from server
+        // Blocker: the script assumes the solvent already exists on the backend, but we probably don't have it.
+        /*
+        try {
```

Comment:
Here, I skip the search because the API does not allow me to yet.

Reply:
You may want to talk to @brili about this because I think he may want you to use an exact search from the API and replace all the material UUID with real UUID.

Reply:
The problem is different here. The commented code is searching material from a "cas" from CRIPT.